pkg/platform/kube/logProxy/elastic/abstract.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2025 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package elastic
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/nuclio/nuclio/pkg/platform/kube/logProxy"
+	"github.com/nuclio/nuclio/pkg/platformconfig"
+
+	"github.com/nuclio/errors"
+	"github.com/nuclio/logger"
+)
+
+type AbstractElasticLikeLogProxy struct {
+	index            string
+	customQueryParam string
+}
+
+// versionInfo is a struct to hold the version information of the Elasticsearch or OpenSearch instance
+type versionInfo struct {
+	Version struct {
+		// OpenSearch-specific
+		Distribution string `json:"distribution"`
+
+		// Both ES and OS have this
+		Number string `json:"number"`
+	} `json:"version"`
+}
+
+// ResolveLogProxy connects to the given endpoint and resolves whether it is an OpenSearch or Elasticsearch instance
+// It returns a LogProxy implementation accordingly
+func ResolveLogProxy(logger logger.Logger, config *platformconfig.ElasticSearchConfig) (logProxy.LogProxy, error) {
```

Comment:
This is essentially a factory, and you create the `LogProxy` instance, not only resolve.
```suggestion
func CreateLogProxy(logger logger.Logger, config *platformconfig.ElasticSearchConfig) (logProxy.LogProxy, error) {
```